[New Term Entry] Adding compareTo() to Java date terms (#2848)
Codecademy/docs: content/java/concepts/date/terms/compareTo/compareTo.md

---
Title: '.compareTo()'
Description: 'Compares two date objects.'
Subjects:
  - 'Code Foundations'
  - 'Computer Science'
Tags:
  - 'Date'
  - 'Methods'
  - 'Comparison'
CatalogContent:
  - 'learn-java'
  - 'paths/computer-science'
---

> The **`Date.compareTo()`** method compares one Date object with another and returns an integer value accordingly.

Reviewer: ```suggestion
The **`Date.compareTo()`** method compares one `Date` object with another and returns an integer value accordingly.
```

## Syntax

```pseudo
thisDate.compareTo(argumentDate)
```

The `Date.compareTo()` takes only one parameter:

- `argumentDate`: A `Date` object to be compared chronologically with the given date.

> It returns

Reviewer: ```suggestion
It returns:
```

> - the value `0` if the argument Date is equal to this Date.

Reviewer: ```suggestion
- The value `0` if `argumentDate` is equal to `thisDate`.
```

Reviewer: Update the other bullet points with the formatting shown above

- a value less than `0` if this Date is before the Date argument.
- a value greater than `0` if this Date is after the Date argument.

> The function throws a single exception that is [NullPointerException](https://www.codecademy.com/resources/docs/java/errors/nullpointerexception) if the argumentDate is null.

Reviewer: ```suggestion
The function throws a single exception, a [NullPointerException](https://www.codecademy.com/resources/docs/java/errors/nullpointerexception), if `argumentDate` is null.
```
